realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -60,26 +63,76 @@
  */
 public class RealmQuery<E extends RealmObject> {
 
-    private final Realm realm;
-    private final Table table;
-    private final LinkView view;
-    private final TableQuery query;
-    private final Map<String, Long> columns;
-    private final Class<E> clazz;
-
+    private BaseRealm realm;
+    private Class<E> clazz;
+    private String className;
+    private Map<String, Long> columns = new HashMap<String, Long>();
```

Comment:
seems no need to initialize here.

Every constructor initializes `columns` again.